File: src/基础算法和其他/模拟/P11004_蓝桥杯2024省Python_B_魔法巡游/Main2.kt
package 基础算法和其他.模拟.P11004_蓝桥杯2024省Python_B_魔法巡游

fun main() {
    val n = readLine()!!.toInt()

    val si = IntArray(n + 1)
    val ti = IntArray(n + 1)

    readLine()!!.split(" ").map { it.toInt() }.forEachIndexed { index, value ->
        si[index + 1] = value
    }

    readLine()!!.split(" ").map { it.toInt() }.forEachIndexed { index, value ->
        ti[index + 1] = value
    }

    var inp = 1
    var count = 0

    for (i in 1..n) {
        val temp = if (inp % 2 == 1) si[i] else ti[i]

        var flag = false
        if (temp == 0) {
            flag = true
        } else {
            var num = temp
            while (num > 0) {
                val digit = num % 10
                if (digit == 0 || digit == 2 || digit == 4) {
                    flag = true
                    break
                }
                num /= 10
            }
        }

        if (flag) {
            count++
            inp++
        }
    }

    println(count)
}

File: src/基础算法和其他/模拟/P11004_蓝桥杯2024省Python_B_魔法巡游/Main2.js
function check(num) {
    if (num === 0) return true;
    let n = num;
    while (n > 0) {
        const digit = n % 10;
        if (digit === 0 || digit === 2 || digit === 4) return true;
        n = Math.floor(n / 10);
    }
    return false;
}

function main() {
    const input = require('fs').readFileSync(0, 'utf-8').split(/\s+/);
    let ptr = 0;
    const n = parseInt(input[ptr++]);

    const si = new Array(n + 1).fill(0);
    const ti = new Array(n + 1).fill(0);

    for (let i = 1; i <= n; i++) si[i] = parseInt(input[ptr++]);
    for (let i = 1; i <= n; i++) ti[i] = parseInt(input[ptr++]);

    let inp = 1, count = 0;
    for (let i = 1; i <= n; i++) {
        const temp = (inp % 2 === 1) ? si[i] : ti[i];
        if (check(temp)) {
            count++;
            inp++;
        }
    }
    console.log(count);
}

main();